### PR TITLE
refactor(homepage): project card refinement

### PR DESCRIPTION
**Pull request by ben196888**
Refine project card design

## Major

- remove project card images and replace them with job avatars

## minor

- remove deprecated style and components
- adjust the layout design on large and medium screen

# Impaction

### Before

Screenshot 2566-10-31 at 20 04 26
Screenshot 2566-10-31 at 20 04 23
Screenshot 2566-10-31 at 20 04 16

### After

Screenshot 2566-10-31 at 22 09 35
Screenshot 2566-10-31 at 20 30 32
Screenshot 2566-10-31 at 20 30 30

### Build and deploy time

| item       | Before | After  |
| ---------- | ------ | ------ |
| w/ Cached  | 00m00s | 00m00s |
| w/o Cached | 00m00s | 00m00s |

### CODE REVIEW

**Comment by ben196888** [comment]
Very nice proof of concept idea. I think we should consider these two methods to make this pull request better.
1. Split `Cards` component into two or three Cards components. Project cards are styling differently from job and event cards. It's nice to have more Cards component to render different card styling
2. Move off the colour mapping logic from client side to server side. The colour of background and border should be provided from CMS items. Hardcoded the colour mapping in the server side and render accordingly in the client side makes migrating colour property to CMS items easier.

**Comment by ben196888** [comment]
The Card component renders ProjectCard and non-ProjectCard differently. The naming of those files are confusing.
I think it would make more sense if we specify a new `ProjectCard` component and make a `DefaultCard` component to support the other card types. 

Please extract out the ProjectCard JSX to be `ProjectCard` component and rename nonProjectCard component to be `DefaultCard` component.

The pseudo code in `Card` component will be something like follows.

```
const Card = (props) => {
  if (props.data.type === 'project_card') {
    return <ProjectCard {...props} />
  }
  return <DefaultCard {...props} />
}
```

**Comment by ljc1991** [comment]
> The Card component renders ProjectCard and non-ProjectCard differently. The naming of those files are confusing. I think it would make more sense if we specify a new `ProjectCard` component and make a `DefaultCard` component to support the other card types.
> 
> Please extract out the ProjectCard JSX to be `ProjectCard` component and rename nonProjectCard component to be `DefaultCard` component.
> 
> The pseudo code in `Card` component will be something like follows.
> 
> ```tsx
> const Card = (props) => {
>   if (props.data.type === 'project_card') {
>     return <ProjectCard {...props} />
>   }
>   return <DefaultCard {...props} />
> }
> ```

Yes, Sir! 🫡

File: homepage/src/components/cards/defaultCard.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import Image from 'next/image';
+import { ParseMarkdownAndHtml } from '../parseMarkdownAndHtml';
+
+const DefaultCard = ({ card }) => (
+  <div className="col-xl-4 col-lg-6 mb-3" id={card.data.id}>
```
Comment (ljc1991) [inline homepage/src/components/cards/defaultCard.jsx:5]: `col-lg-6` will display 1 col when screen width is less than 991 px.
截圖 2023-10-31 15 41 54

I think it should be `col-md-6` to display 2 cols in md screen (≥768px).
截圖 2023-10-31 15 42 09

Comment (ben196888) [reply homepage/src/components/cards/defaultCard.jsx:5]: can you also attach some screenshots of English card page based on the 2 column design? just curious about the wording may be too squeeze in English

Comment (ljc1991) [reply homepage/src/components/cards/defaultCard.jsx:5]: en version
截圖 2023-10-31 15 54 54

截圖 2023-10-31 15 54 46

**Pull request merged**